### DISCUSSION TITLE
qt6ct doesn't detect kvantum in Ubuntu and Fedora

### DISCUSSION THREAD

**Discussion by ghost** [tsujan/Kvantum#941] [Q&A]
Hello, the title is pretty self explanatory, i have already reported the bug on Ubuntu, Fedora and the Fedora qt6ct maintainer on qt6ct's github. I will copypaste the bugreport i made in Fedora as it applies to both distros. 
Something to note, i don't see this bug on Arch Linux and NixOS:

Ubuntu's bug report: https://bugs.launchpad.net/ubuntu/+source/qt6ct/+bug/2054321
Fedora's bug report: https://bugzilla.redhat.com/show_bug.cgi?id=2263614
qt6ct's upstream bug report: https://github.com/trialuser02/qt6ct/issues/40

When you have qt6ct and kvantum installed, what would usually happen is that qt6ct shows kvantum as an available option in the drop-down menu in the "appearance" tab to use to theme QT programs (the correct behavior that qt5ct shows).

Instead, qt6ct simply ignores that kvantum is an option.

Reproducible: Always

Steps to Reproduce:
1) install the "qt6ct" and kvantum "packages"
2) launch qt6ct 
3) try to find the kvantum option in the drop down menu inside qt6ct's appearance tab.
Actual Results:  
No kvantum option appeared in the dropdown menu. 

Expected Results:  
You should see the kvantum option in the dropdown menu.

I suspect that this bug is related specifically to qt6ct because kvantum can theme programs without problems if it's selected as one of the theming options inside KDE's appearance settings. 

This problem is also present on Debian 12 and Ubuntu 23.10, if someone figures out a solution please tell the developers on charge of these packages in their respective distros to fix it. 
With urgency!!! Ubuntu 24.04 LTS is around the corner.

Here is a link that displays the bug graphically, 1 picture is worth more than 1000 words:
https://www.reddit.com/user/RaxelPepi/comments/1an8hxf/qt6ct_bug_example_and_qt5ct_showcasing_intended/

**Comment by tsujan** [comment]
My guess is that Kvantum's Qt6 plugin isn't installed at all in your distro. Arch's package includes both Qt5 and Qt6 plugins of Kvantum. Other distros may have separate packages for them, or they may lack a package for the Qt6 plugin. So, the problem (if any) should be reported to the distro.

Kvantum has supported Qt6 for years, with its default compilation being against Qt5. Kvantum 1.1.0 will be released soon, and the default compilation will switch to Qt6 in it (although it'll support Qt5 too).

**Reply by ghost** [reply]
I searched around the repositories of Ubuntu and Fedora, for the Qt6 Kvantum plugin you mentioned, and your guess looks correct, i couldn't find them anywhere. And after installing Kvantum via Papirus' ppa, i could get qt6ct to recognize Kvantum and theme Qt6 programs.
I will add this information to the bug reports i opened to tell these two distros that their packaging is incomplete, ty for making and maintaining this software.

**Reply by tsujan** [reply]
> I will add this information to the bug reports i opened to tell these two distros that their packaging is incomplete,

That's the correct way of doing it.

Here I develop Kvantum without any special relation to any distro — package building and maintaining is done downstream.

**Comment by principis** [comment]
This was already fixed in Rawhide and is now available in Fedora 39 :slightly_smiling_face: 
https://bugzilla.redhat.com/show_bug.cgi?id=2265209
